lib/rules/newline-before-return.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @fileoverview Rule to check empty line before "return" statement
+ * @author Eric Clemmons <eric@smarterspam.com>
+ * @copyright 2014 Eric Clemmons. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    // Cache line numbers of comments for faster lookup
+    var comments = context.getAllComments().map(function (token) {
+        return token.loc.start.line;
+    });
+
+    //--------------------------------------------------------------------------
+    // Helpers
+    //--------------------------------------------------------------------------
+
+    function hasEmptyLineBefore(node) {
+        var previousToken = context.getTokenBefore(node);
+
+        var lineNum = node.loc.start.line,
+            previousLineNum = previousToken.loc.end.line;
+
+        // Measure the # of lines between tokens
+        var gap = lineNum - previousLineNum - 1;
+
+        // Comments don't count towards the gap
+        while (previousLineNum++ < lineNum) {
+            if (comments.indexOf(previousLineNum) !== -1) {
+                gap--;
+            }
+        }
+
+        // There should be an empty line _somewhere_ between tokens
+        return gap > 0;
+    }
+
+    function isInBlock(node) {
+        return node.parent.type === "BlockStatement";
+    }
+
+    function isFirstStatementInBlock(node) {
+        var previousToken = context.getTokenBefore(node);
+
+        if (!previousToken) {
+            return true;
+        }
+
+        if (!isInBlock(node)) {
+            return false;
+        }
+
+        if (previousToken.type !== "Punctuator") {
+            return false;
+        }
+
+        return previousToken.value === "{";
+    }
+
+    function checkForBlankLine(node) {
+        if (isFirstStatementInBlock(node)) {
```

Comment:
Can you explain this exception in the documentation? 

I'm wondering what should be considered as a statement group. The description of the issue says that a preceding  empty line is not required if the return statement is the first statement of a statement group. Are the child nodes of an if statement considered a statement group?

```js 
if (a) 
    return;
```
Should this example be valid?